### PR TITLE
[ROCm] add options to disable fused attn backend compilation

### PR DESCRIPTION
**Pull request by wangye805**
# Description

Add options to disable rocm fused attn backend compilations

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Infra/Build change
- [ ] Code refractor

## Changes

Now NVTE_FUSED_ATTN, NVTE_FUSED_ATTN_CK, and NVTE_FUSED_ATTN_AOTRITON also controls the compilation

# Checklist:

- [x] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [x] The functionality is complete
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by wenchenvincent — CHANGES_REQUESTED** [review]
Could you update the README about this change? Otherwise, it looks good.

**Comment by wangye805** [comment]
Thanks @wenchenvincent, just updated the README about the usage. Please help take a look at it again.

File: README.rst [hunk 1]
```diff
@@ -220,12 +220,18 @@ If you want to check that only previously tuned algorithms are used by your appl
...
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 Currently ROCm TE supports two backends, AOTriton and CK, for fused attention. 
-To enable specific backends, the following environment variables can be used:
+To enable specific backends in compilation and/or in runtime, the following environment variables can be used:
 
 * NVTE_FUSED_ATTN - enable the fused attention, default = 1;
 * NVTE_FUSED_ATTN_CK - enable the CK backend, default = 1;
 * NVTE_FUSED_ATTN_AOTRITON - enable the AOTriton backend, default = 1.
 
+Setting env NVTE_FUSED_ATTN<BACKEND>=0 in compilation will skip the build of the specific backend, which saves the overall building time.
```
Comment (wenchenvincent) [inline README.rst]: Nit: Change to `NVTE_FUSED_ATTN_<BACKEND>`

Comment (wangye805) [reply README.rst]: Done

**Pull request merged**